grpc-ingest/src/grpc.rs

```diff
@@ -119,97 +131,140 @@ pub async fn run(config: ConfigGrpc) -> anyhow::Result<()> {
-                        exec.push(GrpcJob::ProcessSubscribeUpdate(Box::new(msg)));
+    let subscriptions = config.subscriptions.clone();
+
+    let mut subscription_tasks: Vec<JoinHandle<Result<(), anyhow::Error>>> =
+        Vec::with_capacity(subscriptions.len());
+
+    for subscription in subscriptions {
+        let subscription_label = subscription.0.clone();
+        let subscription_config = Arc::new(subscription.1);
+        let config = Arc::clone(&config);
+        let connection = connection.clone();
+        let pipe = Arc::clone(&pipe);
+        let subscription_task_handle: JoinHandle<Result<(), anyhow::Error>> = tokio::spawn(
+            async move {
+                let mut shutdown = create_shutdown()?;
+                let SubscriptionConfig { stream, filter } = subscription_config.as_ref();
+
+                let stream_config = Arc::new(stream.clone());
+                let mut accounts = HashMap::with_capacity(1);
+                let mut transactions = HashMap::with_capacity(1);
+
+                match filter {
+                    ConfigGrpcRequestFilter::Accounts(accounts_filter) => {
+                        accounts.insert(
+                            subscription_label.clone(),
+                            accounts_filter.clone().to_proto(),
+                        );
                     }
-                    Some(UpdateOneof::Ping(_)) => {
-                        let ping = subscribe_tx
-                            .send(SubscribeRequest {
-                                ping: Some(SubscribeRequestPing { id: PING_ID }),
-                                ..Default::default()
-                            })
-                            .await;
-
-                        match ping {
-                            Ok(_) => {
-                                debug!(target: "grpc2redis", action = "send_ping", message = "Ping sent successfully", id = PING_ID)
-                            }
-                            Err(err) => {
-                                warn!(target: "grpc2redis", action = "send_ping_failed", message = "Failed to send ping", ?err, id = PING_ID)
+                    ConfigGrpcRequestFilter::Transactions(transactions_filter) => {
+                        transactions.insert(
+                            subscription_label.clone(),
+                            transactions_filter.clone().to_proto(),
+                        );
+                    }
+                }
+
+                let request = SubscribeRequest {
+                    accounts,
+                    transactions,
+                    ..Default::default()
+                };
+
+                let mut dragon_mouth_client =
+                    GeyserGrpcClient::build_from_shared(config.geyser_endpoint.clone())?
+                        .x_token(config.x_token.clone())?
+                        .connect_timeout(Duration::from_secs(10))
+                        .timeout(Duration::from_secs(10))
```

Comment:
Lets pull this to top level geyser config.

```
geyser:
  endpoint: ""
  x_token: ""
  connection_timeout:
  timeout:
```

Apply this to all subscriptions.

Reply:
done